lib/do_profile/profilecommon.js

```diff
@@ -135,6 +137,213 @@ function setCurrentProfile(opts, cb) {
+                        profile.url)));
+                } else if (res.statusCode === 401) {
+                    var portalUrl = portalUrlFromCloudapiUrl(profile.url);
+                    if (portalUrl) {
+                        next(new errors.SetupError(err, format(
+                            'invalid credentials. Visit <%s> to create the '
+                            + '"%s" account and/or add your SSH public key',
+                            portalUrl, profile.account)));
+                    } else {
+                        next(new errors.SetupError(err, format(
+                            'invalid credentials. You must create the '
+                            + '"%s" account and/or add your SSH public key',
+                            profile.account)));
+                    }
+                } else if (res.statusCode === 200) {
+                    if (svcs.cmon) {
+                        cmonHost = svcs.cmon;
+                        log.trace({cmonHost: cmonHost},
+                            'profileCmonSetup: checkForCmonService');
+                        next();
+                    } else {
+                        next(new errors.SetupError(err, format(
+                            'no "cmon" service on this datacenter',
+                            profile.url)));
+                    }
+                } else {
+                    // TODO: If this doesn't show res details, add that.
+                    next(new errors.SetupError(err, format(
+                        'unexpected response from cloudapi %s: %s status code',
+                        profile.url, res.statusCode)));
+                }
+            });
+        },
+
+        function mentionSettingUp(arg, next) {
+            console.log('Generating CMON certificates for profile "%s".',
+                profile.name);
+            next();
+        },
+
+        function getSigningKey(arg, next) {
+            var keyId = profile.keyId;
+            var keyFp = sshpk.parseFingerprint(keyId);
+            agent.listKeys(function (err, keys) {
+                if (err) {
+                    next(err);
+                    return;
+                }
+
+                arg.certSignKey = keys.filter(function (k) {
+                    return (keyFp.matches(k));
+                })[0];
+
+                if (!arg.certSignKey) {
+                    next(err);
+                    return;
+                }
+                next();
+            });
+        },
+
+        function generateAndSignCert(arg, next) {
+            var account = profile.account;
+            var fp = sshpk.parseFingerprint(profile.keyId);
+            var subj = sshpk.identityFromDN('CN=' + account);
+            var issuer = sshpk.identityFromDN('CN=' + fp);
+            var certSignKey = arg.certSignKey;
+
+            sslKey = sshpk.generatePrivateKey('ecdsa');
+
+            var certOpts = {
+                lifetime: 24*3600*opts.lifetime,
```

Comment:
Can you move 24 * 3600 to be a constant - e.g. `SECONDS_PER_DAY = 24 * 60 * 60`

Reply:
Done.